internal/kafka/client_test.go

```diff
@@ -5,17 +5,30 @@ package kafka // import "github.com/open-telemetry/opentelemetry-collector-contr
 
 import (
 	"context"
+	"encoding/pem"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 	"time"
 
 	"github.com/IBM/sarama"
+	"github.com/rcrowley/go-metrics"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kfake"
+	"go.opentelemetry.io/collector/config/configopaque"
 	"go.opentelemetry.io/collector/config/configtls"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/kafka/configkafka"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/kafka/kafkatest"
 )
 
+func init() {
+	// Disable the go-metrics registry, as there's a goroutine
+	// leak in the Sarama code that uses it.
+	metrics.UseNilMetrics = true
+}
```

Comment:
I am never a big fan of seeing init statements, is there a way that this could be raised as in issue in the upstream repo to address the leak of the go routine?

Reply:
Once we have the issue created, can you comment it here so we can follow up later on?

Reply:
There was an issue, it went stale: https://github.com/IBM/sarama/issues/1321

My fix would be to swap to franz-go ;)

I'll see if we can work around the bug another way.

Reply:
P.S. this is also the documented by Sarama as the way to disable metrics: https://github.com/IBM/sarama/blob/d6ca80a198c4fbc467c67f7430182eda6881b12c/config.go#L513

So I thought it made sense for the tests, but not non-test code because that might inadvertently disable metrics in other components.

Reply:
The suggestion in https://github.com/IBM/sarama/issues/1321#issuecomment-547530477 to explicitly unregister metrics doesn't help. I can't see a better alternative.

Reply:
Mm, I'd still like a reference to the issue so I can come back to it once I get some free time.

Reply:
:+1: added a comment